code/game/gamemodes/umbrage/umbrage_abilities/demented_outburst.dm

```diff
@@ -0,0 +1,46 @@
+//Emits a shockwave that blasts everyone and everything nearby far away. People close to the user are deafened and stunned.
+/datum/action/innate/umbrage/demented_outburst
+	name = "Demented Outburst"
+	id = "demented_outburst"
+	desc = "Deafens and confuses listeners, and knocks away everyone nearby. Very loud."
+	button_icon_state = "umbrage_demented_outburst"
+	check_flags = AB_CHECK_CONSCIOUS
+	psi_cost = 60 //big boom = big cost
+	lucidity_cost = 2
+	blacklisted = 0
+
+/datum/action/innate/umbrage/demented_outburst/Activate()
+	owner.visible_message("<span class='warning'>[owner] begins to growl!</span>", "<span class='velvet bold'>cap...</span><br>\
+	<span class='danger'>You begin harnessing every ounce of your power...</span>")
+	playsound(owner, 'sound/magic/demented_outburst_charge.ogg', 100, 0)
+	addtimer(CALLBACK(src, .proc/outburst, owner), 50)
+	return TRUE
+
+/datum/action/innate/umbrage/demented_outburst/proc/outburst(mob/living/owner)
+	if(!owner || owner.stat)
+		return
+	owner.visible_message("<span class='boldwarning'>[owner] lets out a deafening scream!</span>", "<span class='velvet bold italics'>WSWU!</span><br>\
+	<span class='danger'>You let out a deafening outburst!</span>")
+	playsound(owner, 'sound/magic/demented_outburst_scream.ogg', 150, 0)
+	var/list/thrown_atoms = list()
+	for(var/turf/T in view(5, owner))
+		for(var/atom/movable/AM in T)
+			thrown_atoms += AM
+	for(var/atom/movable/AM in thrown_atoms)
```

Comment:
Evidently not. I wanted an identical effect, so why rewrite code that already existed? No need to reinvent the wheel.

That said, there are improvements coming to this code in the near future.